raiden_contracts/deploy/etherscan_verify.py

```diff
@@ -41,108 +41,94 @@
+        chain_id: int,
+        apikey: str,
+        guid: Optional[str],
+        contract_name: Optional[str],
+):
     if guid:
-        guid_status(etherscan_api, guid)
+        guid_status(api_of_chain_id(chain_id), guid)
         return
 
     if contract_name is None or contract_name == CONTRACT_ENDPOINT_REGISTRY:
-        path = source_path['raiden'].joinpath(f'{CONTRACT_ENDPOINT_REGISTRY}.sol')
-        source = path.read_text()
-        constructor_args = ''
-        etherscan_verify_contract(
-            etherscan_api,
-            apikey,
-            contract_manager.contracts[CONTRACT_ENDPOINT_REGISTRY],
-            CONTRACT_ENDPOINT_REGISTRY,
-            deployment_info['contracts'][CONTRACT_ENDPOINT_REGISTRY],
-            source,
-            constructor_args,
-        )
+        etherscan_verify_contract(chain_id, apikey, 'raiden', CONTRACT_ENDPOINT_REGISTRY)
 
     if contract_name is None or contract_name == CONTRACT_SECRET_REGISTRY:
-        path = source_path['raiden'].joinpath(f'{CONTRACT_SECRET_REGISTRY}.sol')
-        source = path.read_text()
-        constructor_args = ''
-        etherscan_verify_contract(
-            etherscan_api,
-            apikey,
-            contract_manager.contracts[CONTRACT_SECRET_REGISTRY],
-            CONTRACT_SECRET_REGISTRY,
-            deployment_info['contracts'][CONTRACT_SECRET_REGISTRY],
-            source,
-            constructor_args,
-        )
+        etherscan_verify_contract(chain_id, apikey, 'raiden', CONTRACT_SECRET_REGISTRY)
 
     if contract_name is None or contract_name == CONTRACT_TOKEN_NETWORK_REGISTRY:
-        joined_file = Path(__file__).parent.joinpath('joined.sol')
-        command = [
-            './utils/join-contracts.py',
-            '--import-map',
-            '{"raiden": "contracts", "test": "contracts/test", "services": "contracts/services"}',
-            'contracts/TokenNetworkRegistry.sol',
-            str(joined_file),
-        ]
-        old_working_dir = Path.cwd()
-        chdir(Path(__file__).parent.parent)
-        try:
-            combiner = subprocess.Popen(command)
-            combiner.wait()
-        finally:
-            chdir(old_working_dir)
-
-        source = joined_file.read_text()
-
-        constructor_arguments = deployment_info['contracts'][
-            CONTRACT_TOKEN_NETWORK_REGISTRY
-        ]['constructor_arguments']
-        abi = contract_manager.contracts[CONTRACT_TOKEN_NETWORK_REGISTRY]['abi']
+        etherscan_verify_contract(chain_id, apikey, 'raiden', CONTRACT_TOKEN_NETWORK_REGISTRY)
+
+
+def api_of_chain_id(chain_id: int) -> str:
+    if chain_id == 3:
+        return 'https://api-ropsten.etherscan.io/api'
+    elif chain_id == 4:
+        return 'https://api-rinkeby.etherscan.io/api'
+    elif chain_id == 42:
+        return 'https://api-kovan.etherscan.io/api'
+    elif chain_id == 1:
+        return 'https://api.etherscan.io/api'
+    else:
+        raise ValueError("Unknown chain_id {chain_id}")
```

Comment:
Not sure how important having this specific error message is. But apart from that, this function could be replace by a plain `dict`.

Reply:
That's a good idea.  I'll do it in this PR.